optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -120,9 +125,24 @@ void makeGizmoBeanFactoryUnremovable(BuildProducer<UnremovableBeanBuildItem> unr
         unremovableBeans.produce(UnremovableBeanBuildItem.beanTypes(OptaPlannerGizmoBeanFactory.class));
     }
 
+    @BuildStep(onlyIf = IsDevelopment.class)
+    public DevConsoleRuntimeTemplateInfoBuildItem getSolverConfig(SolverConfigBuildStep solverConfigBuildStep) {
+        SolverConfig solverConfig = solverConfigBuildStep.getSolverConfig();
+        if (solverConfig != null) {
+            StringWriter effectiveSolverConfigWriter = new StringWriter();
+            SolverConfigIO solverConfigIO = new SolverConfigIO();
+            solverConfigIO.write(solverConfig, effectiveSolverConfigWriter);
+            return new DevConsoleRuntimeTemplateInfoBuildItem("solverConfigProperties",
+                    new OptaPlannerDevUIPropertiesSupplier(effectiveSolverConfigWriter.toString()));
+        } else {
+            return new DevConsoleRuntimeTemplateInfoBuildItem("solverConfigProperties",
+                    new OptaPlannerDevUIPropertiesSupplier());
```

Comment:
It seems that this does nothing. That suggests to me that there will be an empty item somewhere in the Dev UI. What happens if we return null instead?

Reply:
Null pointer exception on attempt to access any attribute of `solverConfigProperties`.  In particular, an empty `OptaPlannerDevUIPropertiesSupplier()` creates an  `OptaPlannerDevUIProperties` with the given "solverConfig":

```
<!-- Plugin execution was skipped because there are no @PlanningSolution or @PlanningEntity annotated classes -->
<solver />
```

Additionally, it set the constraint list to the empty list, and the model properties to an empty instance.